internal/playlist_creator/PlaylistWriter.go

```diff
@@ -34,21 +37,27 @@ func WritePlaylist(config *config.Config, fileData *FileData) {
 	for entry := range chunks {
 		playlistNum++
 		fmt.Printf("Writing playlist %d\n", playlistNum)
-		writePlaylistFile(config, playlistNum, &entry)
+		var err = writePlaylistFile(config, playlistNum, &entry)
+		if err != nil {
+			return err
+		}
 	}
+	return nil
 }
 
-func writePlaylistFile(config *config.Config, playlistNum int, fileEntries *[]FileEntry) {
+func writePlaylistFile(config *config.Config, playlistNum int, fileEntries *[]FileEntry) error {
 	file, err := os.Create(config.OutputPath + "/" + config.PlaylistName + "_" + fmt.Sprintf("%02d", playlistNum) + ".m3u8")
 	if err != nil {
-		log.Fatal(err)
+		log.Printf("Could not create playlist file %s_%s.m3u8: %s", config.PlaylistName, fmt.Sprintf("%02d", playlistNum), err.Error())
+		return err
```

Comment:
i've seen you sometimes return the error directly. while i would usually advise to wrap errors, it depends on how detailed you want your errors, so i will leave the direct error returns as is.

but in case you already are creating error messages, it may be worth considering to wrap this error and return it, as it would help in debugging later as the code base grows.
```suggestion
		err = fmt.Errorf("could not create playlist file %s_%02d.m3u8: %w", config.PlaylistName, playlistNum, err)
		log.Printf(err.Error())
		return err